### PR TITLE
C/sort/counting sort.c 

### PR DESCRIPTION
**Pull request by Ayush7614**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by Ayush7614** [comment]
plss review it

**Comment by Ayush7614** [comment]
why this checks have failed

File: C/sort/counting sort.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    printf("The Sorted array is : ");
+    for (i = 1; i <= n; i++)
+        printf("%d ", B[i]);
+}
+/*  End of counting_sort()  */
+ 
+/*  The main() begins  */
+int main()
+{
+    int n, k = 0, A[15], i;
```
Comment (atarax665) [inline C/sort/counting sort.c]: Do not fix the size of the array, declare the array as ```A[n];``` after taking the size of the array as input from the user.

File: C/sort/counting sort.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+# The best complexity for Counting Sort is O(n + k). */
+
+/* C Program to sort Elements Using Counting Sort */
+#include <stdio.h>  
+ 
+/*  Counting sort function  */
+void counting_sort(int A[], int k, int n)
+{
+    int i, j;
+    int B[15], C[100];
```
Comment (atarax665) [inline C/sort/counting sort.c]: Do not fix the size of the arrays, calculate the max size of the count array using some algorithm.

File: C/sort/counting sort.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/* Counting sort is a sorting algorithm that sorts the elements of an array by 
+counting the number of occurrences of each unique element in the array. 
+The count is stored in an auxiliary array and the sorting is done by 
+mapping the count as an index of the 
+auxiliary array.
+# The average time complexity for Counting Sort is O(n + k).
```
Comment (atarax665) [inline C/sort/counting sort.c]: Mention what are ```n``` and ```k``` here.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by atarax665 — COMMENTED** [review]
Update README.md and change the file name from ```counting sort.c``` to ```Counting_sort.c```.

**Pull request closed without merging**